src/middlewares/authenticationMiddleware.ts

```diff
@@ -0,0 +1,34 @@
+import { Response, Request, NextFunction } from 'express'
+import { decodeToken } from '../utils/jwt'
+import { getUserById } from '../services/userServices'
+
+export interface MyExtendedRequest extends Request {
+  userDetails?: object
+}
```

Comment:
```suggestion
export interface MyExtendedRequest extends Request {
  userDetails?: Model<UserAttributes>
}
```
This can be done through .d.ts file so that it can be globally recognized
https://stackoverflow.com/questions/37377731/extend-express-request-object-using-typescript
and instead of calling it userDetails, you can call it, user and make it simple

Reply:
The file `express.d.ts` wasn't working so I did it directly in this file

declare global {
  namespace Express {
    interface Request {
      user?: object
    }
  }
}